src/KafkaFlow.Serializer/SerializerConsumerMiddleware.cs

```diff
@@ -45,15 +45,17 @@ public Task Invoke(IMessageContext context, MiddlewareDelegate next)
                 return next(context);
             }
 
-            if (!(context.Message is byte[] rawData))
+            if (!(context.Message.Value is byte[] rawData))
             {
                 throw new InvalidOperationException(
                     $"{nameof(context.Message)} must be a byte array to be deserialized and it is '{context.Message.GetType().FullName}'");
             }
 
-            context.TransformMessage(this.serializer.Deserialize(rawData, messageType));
-
-            return next(context);
+            return next(
+                context.TransformMessage(
```

Comment:
what do you think about passing just the key and message to 'context.TransformMessage' and creating the Message object just there?

Reply:
Yes, I think it is better too. Changed